### ISSUE TITLE
Enhancement: [switch-exhaustiveness-check] null and undefined catched with default.

### ISSUE THREAD

**Issue by UnderKoen** [typescript-eslint/typescript-eslint#12140]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/switch-exhaustiveness-check/

### Description

I propose an option for switch-exhaustiveness-check when true `null` and `undefined` are matched with the `default`.

As the following seems too verbose.

```ts
const var: 'test' | 'abc' | null | undefined;

switch(var) {
  case 'test':
    //...
  case 'abc':
    //...
  case null:
  case undefined:
  default:
    //...
}
```

### Fail

```TypeScript
const var: 'test' | 'abc' | null | undefined;

switch(var) {
  case 'test':
    //...
  default:
    //...
}
```

### Pass

```TypeScript
const var: 'test' | 'abc' | null | undefined;

switch(var) {
  case 'test':
    //...
  case 'abc':
    //...
  default:
    //...
}
```

**Comment by Hamzaa6296** [comment]
I tested this behavior and can confirm that currently the rule requires explicit handling of **`null`** and **`undefined`**.

I agree that treating them as part of **`default`** could reduce verbosity, but I wonder if this might reduce strictness in some cases. Should this be configurable via an option instead of default behavior?

**Comment by JoshuaKGoldberg** [comment]
In general / most of the time, `null` and `undefined` shouldn't be treated as special types. They're both "just another" constituent (possible type) in your union type.

To reduce verbosity in switch statements, I think [`considerDefaultExhaustiveForUnions`](https://typescript-eslint.io/rules/switch-exhaustiveness-check/#considerdefaultexhaustiveforunions) is what you're looking for. It makes the code snippets you've posted pass the rule without reports. [[playground link for `considerDefaultExhaustiveForUnions`](https://typescript-eslint.io/play/#ts=6.0.2&fileType=.ts&code=CYUwxgNghgTiAEYD2A7AzgF3gNyhAriAFzwDkGImp8APmVAEZjV0r4QS3z4qgBmASxQhgAbgBQ4tAHcBGMAAt4AClwEQASngBvcfERQ0CcpQykie-fAD01gHQPLYQ8cbMLVm-cf7%2BUdhgeVrYOduIAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6ZAd0vzIAt6AHrwCGsZPkoA3RExTJafRGQDW6MAG1w2HImjQA9tEgAaLdqzbskMvqbJKAE10ARRADNR8fAFEho8VKIAGKGAKpMlLaoGPhwiGbYAL4JALpayYlAA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)]

-1 from me on this feature request. I appreciate you filing it though, it's an interesting idea!

**Comment by bradzacher** [comment]
Yeah I'd agree - we have the option and this pattern in of itself is rare. Closing.
